packages/features/ee/round-robin/roundRobinReassignment.ts

```diff
@@ -0,0 +1,364 @@
+import EventManager from "@calcom/core/EventManager";
+import dayjs from "@calcom/dayjs";
+import { sendRoundRobinCancelledEmails, sendRoundRobinScheduledEmails } from "@calcom/emails";
+import { getCalEventResponses } from "@calcom/features/bookings/lib/getCalEventResponses";
+import { ensureAvailableUsers, getEventTypesFromDB } from "@calcom/features/bookings/lib/handleNewBooking";
+import type { IsFixedAwareUser } from "@calcom/features/bookings/lib/handleNewBooking";
+import { scheduleEmailReminder } from "@calcom/features/ee/workflows/lib/reminders/emailReminderManager";
+import { isPrismaObjOrUndefined } from "@calcom/lib";
+import logger from "@calcom/lib/logger";
+import { getLuckyUser } from "@calcom/lib/server";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import { getTimeFormatStringFromUserTimeFormat } from "@calcom/lib/timeFormat";
+import { prisma } from "@calcom/prisma";
+import { WorkflowActions, WorkflowTriggerEvents } from "@calcom/prisma/enums";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+
+const bookingSelect = {
+  uid: true,
+  title: true,
+  startTime: true,
+  endTime: true,
+  userId: true,
+  customInputs: true,
+  responses: true,
+  description: true,
+  location: true,
+  eventTypeId: true,
+  destinationCalendar: true,
+  user: {
+    include: {
+      destinationCalendar: true,
+    },
+  },
+  attendees: true,
+  references: true,
+};
+
+export const roundRobinReassignment = async ({ bookingId }: { bookingId: number }) => {
+  const roundRobinReassignLogger = logger.getSubLogger({
+    prefix: ["roundRobinReassign", `${bookingId}`],
+  });
+
+  let booking = await prisma.booking.findUnique({
+    where: {
+      id: bookingId,
+    },
+    select: bookingSelect,
+  });
+
+  if (!booking) {
+    console.error(`Booking ${bookingId} not found`);
+    throw new Error("Booking not found");
+  }
+
+  if (!booking?.user) {
+    console.error(`No user associated with booking ${bookingId}`);
+    throw new Error("Booking not found");
+  }
+
+  const eventTypeId = booking.eventTypeId;
+
+  if (!eventTypeId) {
+    console.error(`Booking ${bookingId} does not have an event type id`);
+    throw new Error("Event type not found");
+  }
+
+  const eventType = await getEventTypesFromDB(eventTypeId);
+
+  if (!eventType) {
+    console.error(`Event type ${eventTypeId} not found`);
+    throw new Error("Event type not found");
+  }
+
+  eventType.users = eventType.hosts.map((host) => ({ ...host.user, isFixed: host.isFixed }));
+
+  const roundRobinHosts = eventType.hosts.filter((host) => !host.isFixed);
```

Comment:
We should probably also still support `eventType.users` as we do in `handleNewBooking` (Also in all places below where eventType.hosts ist used)